sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/UnorderedDistinctMap.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.models.Resource;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.nio.charset.Charset;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.HashSet;
+
+public class UnorderedDistinctMap extends DistinctMap {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper().configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);;
+    private final HashSet<String> resultSet;
```

Comment:
if `resultSet` can be accessed concurrently you should consider making this ConcurrentHashSet

Reply:
Yes, made it a ConcurrentHashSet